### PR TITLE
Upgrade to scassandra 1.0.3.  Change CAS tests to use Scassandra.

### PR DESCRIPTION
**Pull request by tolbertam**
Upgrades to Scassandra 1.0.3 (needed for scassandra/scassandra-server#143) and replaces existing CASRetryPolicyIntegrationTest with tests using Scassandra.

### CODE REVIEW

**Comment by tolbertam** [comment]
looking good in CI

File: driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java [hunk 1]
```diff
@@ -100,14 +102,14 @@ public void should_not_start_speculative_execution_if_first_execution_retries_bu
         // will retry once on this node:
         scassandras.node(1).primingClient().prime(PrimingRequest.queryBuilder()
                         .withQuery("mock query")
-                        .withConsistency(PrimingRequest.Consistency.TWO)
-                        .withResult(PrimingRequest.Result.read_request_timeout)
+                .withConsistency(Consistency.TWO)
+                .withResult(Result.read_request_timeout)
```
Comment (adutra) [inline driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java:106]: There seems to be some indentation issues here.

File: driver-core/src/test/java/com/datastax/driver/core/policies/DefaultRetryPolicyIntegrationTest.java [hunk 2]
```diff
@@ -259,4 +260,25 @@ public void should_try_next_host_on_connection_error(ClosedConnectionConfig.Clos
         assertQueried(2, 1);
         assertQueried(3, 1);
     }
+
+    @Test(groups = "short")
+    public void should_rethrow_on_unavailable_if_CAS() {
```
Comment (adutra) [inline driver-core/src/test/java/com/datastax/driver/core/policies/DefaultRetryPolicyIntegrationTest.java:265]: Awesome!

**Comment by adutra** [comment]
:100:

File: driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java (hunk 1 above)
Comment (tolbertam) [inline driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java:106]: Ah good catch!  I turned off automatic save actions when switching to Scassandra and must have not turned it back on soon enough, I'll fix all of this and squash.

Comment (tolbertam) [inline driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java:106]: Weird, even with save actions turned on it still doesn't format it.  Must be some edge case, in any case i'll fix it.

**Comment by tolbertam** [comment]
Fixed formatting issue in SpeculativeExecutionTest (was in a number of places too in that file, so made it consistent).  This should be ready to go!

**Comment by olim7t** [comment]
LGTM :+1:
